main.go

```diff
@@ -39,21 +40,46 @@ func main() {
 
 	deathNote := make(map[string]bool)
 
-	connected := make(chan bool)
-	disconnected := make(chan bool)
+	firstConnected := make(chan bool, 1)
+	var wg sync.WaitGroup
```

Comment:
algorithm:

1. create a `WaitGroup`
2. every new connection (handled in a new goroutine) increases it by `1`
3. Every closed connection sleeps for 10s and then decrements `WaitGroup` by `1`
4. The main goroutine waits for `WaitGroup` to become `0` (all connections are closed)

The 10 second delay is "reconnect time", so that if a new connection comes in during this delay, `WaitGroup` will be incremented and the main goroutine will continue waiting